### PR TITLE
R4R:add the delegation,unbonding,rewards info to address detail 

### PR DESCRIPTION
**Pull request by weichang-bianjie**

### CODE REVIEW

File: backend/service/service_account.go [hunk 1]
```diff
@@ -36,6 +41,17 @@ func (service *AccountService) Query(address string) (result vo.AccountVo) {
...
+		prefix, _, _ = utils.DecodeAndConvert(valaddress)
+
+	}
+	if prefix == conf.Get().Hub.Prefix.ValAddr {
+		validator, err := document.Validator{}.QueryValidatorDetailByOperatorAddr(valaddress)
+		if err == nil {
+			result.Status = strconv.Itoa(validator.Status)
+			result.Moniker = validator.Description.Moniker
+			result.OperatorAddress = valaddress
+		}
```
Comment (kaifei-bianjie) [inline backend/service/service_account.go]: remove to `else` struct

File: backend/service/service_account.go [hunk 2]
```diff
@@ -84,3 +100,87 @@ func isProfiler(address string) bool {
...
+func (service *AccountService) QueryDelegations(address string) (result []*vo.AccountDelegationsVo) {
+	delegations := lcd.GetDelegationsByDelAddr(address)
+	result = make([]*vo.AccountDelegationsVo, 0, len(delegations))
+	for _, val := range delegations {
+		data := vo.AccountDelegationsVo{
+			Address: val.ValidatorAddr,
+			Shares:  val.Shares,
+			Height:  val.Height,
+		}
+		valdator, err := document.Validator{}.QueryValidatorDetailByOperatorAddr(val.ValidatorAddr)
```
Comment (kaifei-bianjie) [inline backend/service/service_account.go]: don't execute db query in a loop, you can query validators by operator address list

File: backend/service/service_account.go [hunk 3]
```diff
@@ -84,3 +100,87 @@ func isProfiler(address string) bool {
...
+	unbondingdelegations := lcd.GetUnbondingDelegationsByDelegatorAddr(address)
+
+	for _, val := range unbondingdelegations {
+		data := vo.AccountUnbondingDelegationsVo{
+			Address: val.ValidatorAddr,
+			EndTime: val.MinTime,
+			Height:  val.CreationHeight,
+			Amount:  utils.ParseCoin(val.Balance),
+		}
+		valdator, err := document.Validator{}.QueryValidatorDetailByOperatorAddr(val.ValidatorAddr)
```
Comment (kaifei-bianjie) [inline backend/service/service_account.go]: don't execute db query in a loop, you can query validators by operator address list

File: backend/service/service_account.go [hunk 4]
```diff
@@ -84,3 +100,87 @@ func isProfiler(address string) bool {
...
+	if err != nil {
+		logger.Error("GetDistributionRewardsByValidatorAcc have error", logger.String("err", err.Error()))
+		return
+	}
+
+	result.CommissionRewards = commissionrewards
+	result.TotalRewards = rewards
+	for _, val := range delegationrewards {
+		data := vo.DelagationsRewards{Address: val.Validator, Amount: val.Reward}
+		valdator, err := document.Validator{}.QueryValidatorDetailByOperatorAddr(val.Validator)
```
Comment (kaifei-bianjie) [inline backend/service/service_account.go]: don't execute db query in a loop, you can query validators by operator address list

File: backend/rest/controller/tx.go [hunk 5]
```diff
@@ -63,6 +64,12 @@ func registerQueryTxList(r *mux.Router) error {
 		if status != "" {
 			query["status"] = status
 		}
+		if address != "" {
+			query["$or"] = []bson.M{
```
Comment (kaifei-bianjie) [inline backend/rest/controller/tx.go]: optimize suggestion：only `Transfer` tx query address by `from` or `to`, other tx type query address by `from`

File: backend/service/service_account.go (hunk 2 above)
Comment (weichang-bianjie) [reply backend/service/service_account.go]: have done

File: backend/service/service_account.go [hunk 6]
```diff
@@ -35,9 +38,20 @@ func (service *AccountService) Query(address string) (result vo.AccountVo) {
 			}
 			result.Amount = amount
 		}
-		result.Deposits = delegatorService.GetDeposits(address)
+		//result.Deposits = delegatorService.GetDeposits(address)
+		valaddress := utils.Convert(conf.Get().Hub.Prefix.ValAddr, address)
+		prefix, _, _ = utils.DecodeAndConvert(valaddress)
+		if prefix == conf.Get().Hub.Prefix.ValAddr {
```
Comment (kaifei-bianjie) [inline backend/service/service_account.go]: there is no need to get prefix of valaddress

Comment (weichang-bianjie) [reply backend/service/service_account.go]: have done

**Pull request merged**